yarn-project/world-state/src/native/native_world_state.test.ts

```diff
@@ -99,6 +99,311 @@ describe('NativeWorldState', () => {
 
       // initial header should still work as before
       expect(fork.getInitialHeader()).toEqual(initialHeader);
+
+      await fork.close();
+    });
+  });
+
+  describe('Pending and Proven chain', () => {
+    let ws: NativeWorldStateService;
+
+    const compareChains = async (left: MerkleTreeReadOperations, right: MerkleTreeReadOperations) => {
+      expect(await left.getTreeInfo(MerkleTreeId.NULLIFIER_TREE)).toEqual(
+        await right.getTreeInfo(MerkleTreeId.NULLIFIER_TREE),
+      );
+      expect(await left.getTreeInfo(MerkleTreeId.ARCHIVE)).toEqual(await right.getTreeInfo(MerkleTreeId.ARCHIVE));
+      expect(await left.getTreeInfo(MerkleTreeId.L1_TO_L2_MESSAGE_TREE)).toEqual(
+        await right.getTreeInfo(MerkleTreeId.L1_TO_L2_MESSAGE_TREE),
+      );
+      expect(await left.getTreeInfo(MerkleTreeId.NOTE_HASH_TREE)).toEqual(
+        await right.getTreeInfo(MerkleTreeId.NOTE_HASH_TREE),
+      );
+      expect(await left.getTreeInfo(MerkleTreeId.PUBLIC_DATA_TREE)).toEqual(
+        await right.getTreeInfo(MerkleTreeId.PUBLIC_DATA_TREE),
+      );
+
+      expect(await left.getSiblingPath(MerkleTreeId.NULLIFIER_TREE, 0n)).toEqual(
+        await right.getSiblingPath(MerkleTreeId.NULLIFIER_TREE, 0n),
+      );
+      expect(await left.getSiblingPath(MerkleTreeId.ARCHIVE, 0n)).toEqual(
+        await right.getSiblingPath(MerkleTreeId.ARCHIVE, 0n),
+      );
+      expect(await left.getSiblingPath(MerkleTreeId.L1_TO_L2_MESSAGE_TREE, 0n)).toEqual(
+        await right.getSiblingPath(MerkleTreeId.L1_TO_L2_MESSAGE_TREE, 0n),
+      );
+      expect(await left.getSiblingPath(MerkleTreeId.NOTE_HASH_TREE, 0n)).toEqual(
+        await right.getSiblingPath(MerkleTreeId.NOTE_HASH_TREE, 0n),
+      );
+      expect(await left.getSiblingPath(MerkleTreeId.PUBLIC_DATA_TREE, 0n)).toEqual(
+        await right.getSiblingPath(MerkleTreeId.PUBLIC_DATA_TREE, 0n),
+      );
+    };
+
+    const compareSnapshots = async (left: MerkleTreeReadOperations, right: MerkleTreeReadOperations) => {
```

Comment:
Isn't this the same code as compareChains above? Either way, I'd move this to test utils, and loop over MerkleTreeId to make it more manageable.

Reply:
Done